### ISSUE TITLE
JRuby Upgrade

### ISSUE THREAD

**Issue by sam** [sam/harbor#9]
Harbor should be upgraded to support the JRuby VM exclusively. This gives us the flexibility to code to a specific web-server (right now I'm in favor of Jetty), give that "old school Rails" flavor of being able to just run `harbor server` and have an environment up, not having to trouble yourself with the _web server flavor of the week_.

Additionally this would let us do fancy/amazing things like:
- In-process compression of zip archives writing directly to socket without delay or blocking the server (multi-threading FTW)
- Writing a Response directly to the socket without the use of buffering through Rack, resulting in a much more responsive feel even on long requests
- Optional in-process FTP integration allowing you to use bcrypt and programmatically control access, file requests, etc within your application

Basically a lot of things that could really move the State of the Art in Server-Side Ruby web applications forward.

The downside is you give up Rack (though it might be possible to preserve compatibility with existing middleware?), but the upside is you make for a simpler development environment that's ready for scale and efficient resource usage right out of the box.

**Comment by sam** [comment]
Harbor is currently JRuby _compatible_.

Need to add a `warn` and `exit` to ensure it's _required_.

**Comment by fgrehm** [comment]
Could we at least try to have 1.9 **and** "jruby optimized" support?

**Comment by sam** [comment]
The web-server portion wouldn't be too hard to add a YARV compatibility stub I imagine. You could preserve Rack there. Maybe `require "harbor/mri"` or something.

We just can't refer to Rack in the rest of the libraries. It's not used in many places anyway though, mostly a few helpers.

Having a Locale interface that conforms to `java.util.Locale` is pretty trivial, so that could be patched in when you make that require as well.

I think JRuby is the future. :-) But I wouldn't try to actively break MRI/YARV if you want to patch those things in. I'll avoid the `exit`.

**Comment by fgrehm** [comment]
I'll give it try ;-)
